src/main/java/io/cryostat/net/web/http/api/v2/CertificatePostHandler.java

```diff
@@ -151,16 +151,18 @@ public boolean isOrdered() {
             throw new ApiException(409, filePath.toString() + " Certificate already exists");
         }
 
-        try (InputStream fis = fs.newInputStream(certPath);
-                FileOutputStream out = outputStreamFunction.apply(filePath.toFile())) {
-
+        try (InputStream fis = fs.newInputStream(certPath)) {
             Collection<? extends Certificate> certificates = certValidator.parseCertificates(fis);
-            Iterator<? extends Certificate> it = certificates.iterator();
-            while (it.hasNext()) {
-                Certificate certificate = (Certificate) it.next();
-                byte[] buf = certificate.getEncoded();
-                out.write(buf);
+
+            try (FileOutputStream out = outputStreamFunction.apply(filePath.toFile())) {
+                Iterator<? extends Certificate> it = certificates.iterator();
```

Comment:
This predates your patch but since we're here, is there a particular reason to use a while-loop and an Iterator here, rather than a more conventional for-loop/for-each-loop?

Reply:
Personally, I'd prefer a for-each loop. I can refactor it